src/frontend/src/CustomNodes/GenericNode/components/NodeOutputfield/index.tsx

```diff
@@ -0,0 +1,254 @@
+  getGroupOutputNodeId,
+  scapedJSONStringfy,
+} from "../../../../utils/reactflowUtils";
+import {
+  classNames,
+  cn,
+  isThereModal,
+  logHasMessage,
+  logTypeIsError,
+  logTypeIsUnknown,
+} from "../../../../utils/utils";
+import OutputComponent from "../OutputComponent";
+import HandleRenderComponent from "../handleRenderComponent";
+import OutputModal from "../outputModal";
+
+export default function NodeOutputField({
+  selected,
+  data,
+  title,
+  id,
+  colors,
+  tooltipTitle,
+  showNode,
+  index,
+  type,
+  outputName,
+  outputProxy,
+}: NodeOutputFieldComponentType): JSX.Element {
+  const ref = useRef<HTMLDivElement>(null);
+  const nodes = useFlowStore((state) => state.nodes);
+  const edges = useFlowStore((state) => state.edges);
+  const setNode = useFlowStore((state) => state.setNode);
+  const myData = useTypesStore((state) => state.data);
+  const updateNodeInternals = useUpdateNodeInternals();
+  const setFilterEdge = useFlowStore((state) => state.setFilterEdge);
+  const [openOutputModal, setOpenOutputModal] = useState(false);
+  const flowPool = useFlowStore((state) => state.flowPool);
+
+  let flowPoolId = data.id;
+  let internalOutputName = outputName;
+
+  if (data.node?.flow && outputProxy) {
+    const realOutput = getGroupOutputNodeId(
+      data.node.flow,
+      outputProxy.name,
+      outputProxy.id,
+    );
+    if (realOutput) {
+      flowPoolId = realOutput.id;
+      internalOutputName = realOutput.outputName;
+    }
+  }
+
+  const flowPoolNode = (flowPool[flowPoolId] ?? [])[
+    (flowPool[flowPoolId]?.length ?? 1) - 1
+  ];
+
+  const displayOutputPreview =
+    !!flowPool[flowPoolId] &&
+    logHasMessage(flowPoolNode?.data, internalOutputName);
+
+  const unknownOutput = logTypeIsUnknown(
+    flowPoolNode?.data,
+    internalOutputName,
+  );
+  const errorOutput = logTypeIsError(flowPoolNode?.data, internalOutputName);
+
+  if (outputProxy) {
+    console.log(logHasMessage(flowPoolNode?.data, internalOutputName));
+  }
+
+  const preventDefault = true;
+
+  function handleOutputWShortcut() {
+    if (!displayOutputPreview || unknownOutput) return;
+    if (isThereModal() && !openOutputModal) return;
+    if (selected) {
+      setOpenOutputModal((state) => !state);
+    }
+  }
+
+  const output = useShortcutsStore((state) => state.output);
+  useHotkeys(output, handleOutputWShortcut, { preventDefault });
+
+  let disabledOutput =
+    edges.some((edge) => edge.sourceHandle === scapedJSONStringfy(id)) ?? false;
+
+  function renderTitle() {
+    return (
+      <OutputComponent
+        proxy={outputProxy}
+        idx={index}
+        types={type?.split("|") ?? []}
+        selected={
+          data.node?.outputs![index].selected ??
+          data.node?.outputs![index].types[0] ??
+          title
+        }
+        nodeId={data.id}
+        frozen={data.node?.frozen}
+        name={title ?? type}
+      />
+    );
+  }
+  const handleUpdateOutputHide = (value?: boolean) => {
```

Comment:
maybe we can transform this into a hook to update values in the output, just as the HandleOnNewValue

Reply:
sure, but i recommend do it in other pr, this is already too big